test/specs/js-file.js

```diff
@@ -162,6 +162,94 @@ describe('modules/js-file', function() {
+                    'var z = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(file.isEnabledRule('validateQuoteMarks', 3));
+            });
+
+            it('should work with no space before jscs:ignore', function() {
+                var file = createJsFile([
+                    'var x = "1";',
+                    'var y = "1"; //jscs: ignore validateQuoteMarks',
+                    'var z = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(file.isEnabledRule('validateQuoteMarks', 3));
+            });
+
+            it('should work without a space before `jscs`', function() {
+                var file = createJsFile([
+                    'var x = "1";',
+                    'var y = "1"; //jscs: ignore validateQuoteMarks',
+                    'var z = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(file.isEnabledRule('validateQuoteMarks', 3));
+            });
+
+            it('should not re-enable rules', function() {
+                var file = createJsFile([
+                    'var a = "1";',
+                    '// jscs: disable validateQuoteMarks',
+                    'var b = "1"; // jscs: ignore validateQuoteMarks',
+                    'var c = "1";',
+                ].join('\n'));
+                assert(file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(!file.isEnabledRule('validateQuoteMarks', 3));
+                assert(!file.isEnabledRule('validateQuoteMarks', 4));
+            });
+
+            it('should also work with multiple rules', function() {
+                var file = createJsFile([
+                    '// jscs: disable validateQuoteMarks',
+                    'var a = "1"; // jscs: ignore validateQuoteMarks, anotherRule',
+                    'var b = "1";',
+                ].join('\n'));
+                assert(!file.isEnabledRule('validateQuoteMarks', 1));
+                assert(!file.isEnabledRule('validateQuoteMarks', 2));
+                assert(!file.isEnabledRule('validateQuoteMarks', 3));
+                assert(file.isEnabledRule('anotherRule', 1));
+                assert(!file.isEnabledRule('anotherRule', 2));
+                assert(file.isEnabledRule('anotherRule', 3));
+            });
+
+            it('requires a rule to be specified', function() {
+                var file = createJsFile([
+                    '// jscs: disable validateQuoteMarks',
+                    'var a = "1"; // jscs: ignore', // invalid
```

Comment:
Not easily with how disabling all rules is currently represented in the disabled rule index. There's no good way to know which rules to reenable after the line.

The existing comments's behaviour is also a little funny here... I think this would be a good place for refactoring- having the 'all' comments actually add an entry for every rule rather than this special `'*'` rule.